### PR TITLE
Add support for multiple styling solutions in hero

### PR DESCRIPTION
**Pull request by benoitgrelard**

### CODE REVIEW

File: components/MDXComponents.tsx [hunk 1]
```diff
@@ -137,22 +139,13 @@ export const components = {
       {...props}
     />
   ),
-  pre: ({ children }) => <>{children}</>,
-  code: ({ className, hero, showLineNumbers, collapsed, scrollable, line, ...props }) => {
+  pre: PreWithCopyButton,
+  code: ({ className, line, ...props }) => {
     const isInlineCode = !className;
```
Comment (andy-hook) [inline components/MDXComponents.tsx:144]: Should we leave a comment on this for our future selves?

File: components/ThemeToggle.tsx [hunk 2]
```diff
@@ -13,16 +13,6 @@ export const ThemeToggle = (props) => {
         onClick={() => {
           const newTheme = theme === 'dark' ? 'light' : 'dark';
 
-          // When switching themes, you might see demo iframes lagging behind the primitives homepage for a few frames.
-          // This is because `setTheme` hook depends on React component lifecycle to update things. We are toggling
-          // styles manually so that the iframes and the parent page theme change is perfectly synced.
-          document.querySelectorAll('[data-demo-iframe]').forEach((iframe: HTMLIFrameElement) => {
```
Comment (andy-hook) [inline components/ThemeToggle.tsx:19]: Legacy?

File: pages/_app.tsx [hunk 3]
```diff
@@ -55,51 +56,29 @@ function App({ Component, pageProps }: AppProps) {
 
   const isPrimitivesDocs = router.pathname.includes('/docs/primitives');
   const isColorsDocs = router.pathname.includes('/docs/colors');
-  const isIFrame = router.pathname.includes('/iframe');
-  const isDocsPage = isPrimitivesDocs || isColorsDocs || isIFrame;
 
-  if (isIFrame) {
```
Comment (andy-hook) [inline pages/_app.tsx:61]: legacy as well? did we originally want the demos to support dark mode?

**Review by andy-hook — APPROVED** [review]
❤️

File: components/HeroCodeBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    react: 'latest',
+    'react-dom': 'latest',
+    '@radix-ui/colors': 'latest',
+    '@radix-ui/react-icons': 'latest',
+    [`@radix-ui/react-${componentName}`]: 'latest',
+  };
+
+  if (stylingSolution === 'stitches') {
+    dependencies['@stitches/react'] = 'latest';
+  }
```
Comment (andy-hook) [inline components/HeroCodeBlock.tsx:198]: We should add `classnames` as a vanilla css dep

File: components/ThemeToggle.tsx (hunk 2 above)
Comment (benoitgrelard) [reply components/ThemeToggle.tsx:19]: Yeah I guess at some point we had demos in iframe, can't find any reference of it though.,

File: components/MDXComponents.tsx (hunk 1 above)
Comment (benoitgrelard) [reply components/MDXComponents.tsx:144]: yes

**Pull request merged**